main/sceneBuilder.cpp

```diff
@@ -0,0 +1,75 @@
+#include "gtest/gtest.h" // https://google.github.io/styleguide/cppguide.html#Namespaces
+#include "main/sceneBuilder.h"
+#include "obj/generators.h"
+#include "obj/objects.h"
+
+using core::Ray;
+using core::RayHitData;
+using core::Vec3;
+using generators::PointSource;
+using objects::Object;
+using objects::TriangleObj;
+
+const double kSkipFreq = 1000;
+generators::FakeRandomGen fakeRand;
+
+TEST(SceneBuildertest, IsTestSampleRightShape)
+{
+    Vec3 k1(-0.5, -0.5, 0);
+    Vec3 k2(0.5, -0.5, 0);
+    Vec3 k3(-0.5, 0.5, 0);
+    Vec3 k4(0.5, 0.5, 0);
+    Vec3 k5(123, 123, 1);
+
+    SceneBuilder builder;
+
+    const std::vector<TriangleObj> referenceSample = {TriangleObj(k1, k2, k3), TriangleObj(k2, k3, k4)};
+
+    // because how operator== in triangleObj is implemented, I can do that:
+
+    std::vector<std::unique_ptr<Object>> testSample = builder.BuildTestSample();
+    ASSERT_TRUE(std::find(referenceSample.cbegin(), referenceSample.cend(), *testSample[0]) != referenceSample.cend() &&
+                std::find(referenceSample.cbegin(), referenceSample.cend(), *testSample[1]) != referenceSample.cend())
+        << "Triangle1: " << *testSample[0] << ", Triangle2: " << *testSample[1];
+
+    const std::vector<TriangleObj> invalidTestSample = {TriangleObj(k1, k2, k3), TriangleObj(k2, k3, k5)};
+    ASSERT_FALSE(std::find(invalidTestSample.cbegin(), invalidTestSample.cend(), *testSample[0]) != invalidTestSample.cend() &&
```

Comment:
Why didn't you write a function that would do this for you, instead of copying the code?

This is quite a long condition so it is better to get it right.

Reply:
noted. Added to my TODOs